src/directives/infinite-scroll/infinite-scroll.directive.ts

```diff
@@ -314,7 +320,13 @@ export class InfiniteScrollDirective<T = any> implements OnInit, AfterContentIni
 
             const observable = Array.isArray(loadResult) ? of(loadResult) : from(loadResult);
 
-            const subscription = observable.pipe(first()).subscribe(
+            let subscription: Subscription;
+
+            // subscription needs to be a let here if subscription completes right away the complete function can be called
+            // before the assignment. While browsers will ignore this currently as we transpile
+            // all const/let statements to var, when this is no longer the case (or we are running in a test environment)
+            // const will be used and this can throw an error if we try to access a const variable before it is assigned
+            subscription = observable.pipe(first()).subscribe(
```

Comment:
Just looking here we have a case when the subscription complete function is fired right away, however below this we are still adding the subscription to the list of subscriptions: 

```
this._subscriptions.push(subscription);
```

We should probably only add it to the list if it has not already completed

Reply:
So to fix this perhaps (I haven't tested this). But you could create something a variable like this:

```typescript
let completed: boolean = false;
let subscription: Subscription;

...

() => {
    // remove this request from the list
    this._subscriptions = this._subscriptions.filter(s => s !== subscription);

    completed = true;
}

...

if (!completed) {
    this._subscriptions.push(subscription);
}
```

Reply:
Ill try the above and run the tests again.